pkg/tfgen/generate_schema.go

```diff
@@ -417,6 +417,11 @@ func (g *schemaGenerator) genProperty(mod string, prop *variable, pyMapCase bool
 		secret = *prop.info.Secret
 	}
 
+	var replaceOnChange bool
+	if !prop.optional() {
```

Comment:
Really! `optional` tells you this? That's really confusing naming.

Reply:
this is a helper func that checks to see if there is a TF based ForceNew OR a Pulumi override for ForceNew... I agree that it's confusing but that's the func

Reply:
:thinking: This doesn't seem right.  Why would a require property by definition be replace on changes?

Reply:
d'oh! You are correct.... fixing now - thank you both

Reply:
But also - why are we even trying to automatically emit relaceOnChanges?  That is meant to be a way for users to override the provider’s default behaviour, which will be picked up automatically as part of normal provider behaviour.  I’m pretty nervous about using it in this way - feels like a nontrivial fundamental change.

Reply:
I think this started because @susanev wanted a way to query schemas to know what properties would cause a replace on change, rather than it just being text in "description" (https://github.com/pulumi/pulumi-terraform-bridge/issues/458). And I guess some wires have got crossed and confused about what relaceOnChanges in the schema means.

I would of thought that relaceOnChanges being an output of the provider would be a description like what Susan is after, but your comment suggests that it's an input to the provider to override it's behavior?